public/js/src/module/feedback-bar.js

```diff
@@ -0,0 +1,62 @@
+export default {
+    /**
+     * @return {Element}
+     */
+    get feedbackBar() {
+        if (!this._fbBar) {
+            this._fbBar = document.querySelector('#feedback-bar');
+        }
+        return this._fbBar;
+    },
+    /**
+     * @return {NodeList}
+     */
+    get messages() {
+        return this.feedbackBar.querySelectorAll('p');
+    },
+    setCloseHandler() {
+        this.feedbackBar
+            .querySelector('.close')
+            .addEventListener('click', () => {
+                this.hide();
+            });
+    },
+    /**
+     * Shows an unobtrusive feedback bar to the user.
+     *
+     * @param { string } message - message to show
+     * @param {number=} duration - duration in seconds for the message to show
+     */
+    show(message, duration) {
+        // Max 2 messages displayed
+        this.feedbackBar.classList.add('feedback-bar--show');
+        const { messages } = this;
+        if (messages.length > 1) {
+            messages[1].remove();
+        }
+
+        // If an already shown message isn't exactly the same
+        if (!messages[0] || messages[0].innerHTML !== message.trim()) {
+            const newMessage = document.createElement('p');
+            newMessage.innerHTML = message.trim();
+            this.feedbackBar.prepend(newMessage);
+
+            // Automatically remove newly added feedback after a period
+            if (duration) {
```

Comment:
I got a little lost trying to understand (from the function signature) what would happen if the optional `duration` was omitted, or whether it should be optional in the first place. Then of course I came back and looked at this more closely. So the behavior is that it will leave the feedback open until dismissed. This behavior makes sense. But I can't find any call paths which would trigger it. No change needed, just curious if that's something you anticipate?

Reply:
> But I can't find any call paths which would trigger it. 

Yes, you're right. It's actually something we do in a fork. And this ability helps reduce merge conflicts. That's why I added the bonus refactoring and tests ;). Thanks!